### PR TITLE
Added Label Smoothing

### PR DESCRIPTION
**Pull request by DevSinghSachan**
Hi Graham,

I have created this new branch in order to add the label smoothing code as there were a few merge conflicts in the older branch.

### CODE REVIEW

File: xnmt/decoder.py [hunk 1]
```diff
@@ -119,13 +120,32 @@ def get_scores(self, mlp_dec_state):
...
+
     else:
-      return dy.pickneglogsoftmax_batch(scores, ref_action)
+      log_prob = dy.log_softmax(scores)
+      if not xnmt.batcher.is_batched(ref_action):
+        pre_loss = -dy.pick(log_prob, ref_action)
+      else:
+        pre_loss = -dy.pick_batch(log_prob, ref_action)
+
+      pre_loss = dy.mean_batches(pre_loss)
```
Comment (neubig) [inline xnmt/decoder.py]: Why are you doing `mean_batches` here? I think you shouldn't do anything with respect to batches.

Comment (DevSinghSachan) [reply xnmt/decoder.py]: actually, I got confused here and thought that the loss needs to be a scalar 1D value, so was doing mean_batches. I now realize that the loss needs to be present for every row.

I have updated the code now.

**Comment by DevSinghSachan** [comment]
Hi Graham,
Removed the mean_batches() part, got confused earlier as to whether the loss needs to be 1D number or not.
